frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/builders/MigrationsModelToEntityBuilder.java

```diff
@@ -28,12 +28,13 @@ public void build(S source, D destination) {
     private void buildParallelMigrations(S source, D destination) {
         // null value is not applicable for clusters but it should never occur there
         final ParallelMigrationsType parallelMigrationsType = source.getParallelMigrationsType().getSelectedItem();
-        final Integer parallelMigrations =
-            parallelMigrationsType == null ? null :
-            parallelMigrationsType == ParallelMigrationsType.AUTO ? -2 :
-            parallelMigrationsType == ParallelMigrationsType.AUTO_PARALLEL ? -1 :
-            parallelMigrationsType == ParallelMigrationsType.DISABLED ? 0 :
-            source.getCustomParallelMigrations().getEntity();
+        final Integer parallelMigrations = parallelMigrationsType == null ? null
+                : parallelMigrationsType == ParallelMigrationsType.AUTO ? ParallelMigrationsType.AUTO.getValue()
+                        : parallelMigrationsType == ParallelMigrationsType.AUTO_PARALLEL
+                                ? ParallelMigrationsType.AUTO_PARALLEL.getValue()
+                                : parallelMigrationsType == ParallelMigrationsType.DISABLED
+                                        ? ParallelMigrationsType.DISABLED.getValue()
+                                        : source.getCustomParallelMigrations().getEntity();
```

Comment:
Maybe `switch` will look better here? Like you've done in `MigrationOptionsMapper`.

Reply:
Better now?

Reply:
Yes, much better, thanks.